### ISSUE TITLE
Serilog doesn't log params if at least one of them is not presented in message template

### ISSUE THREAD

**Issue by Alfredovec** [serilog/serilog#1180]
Hello dear Serilog team, 

**Does this issue relate to a new *feature* or an existing *bug*?**
- [x] Seems like a bug

**What version of Serilog is affected?  Please list the related NuGet package.**
- [x] 2.6.0

**What is the target framework and operating system? See [target frameworks](https://docs.microsoft.com/en-us/nuget/schema/target-frameworks) & [net standard matrix](https://docs.microsoft.com/en-us/dotnet/standard/net-standard).**

- [x] netCore 2.0

**Please describe the current behavior?**

Case 1: 
Log.Debug("Hello World", new { Param1 = "Test1" }, new { Param2 = "Test2" })
// Doesn't log neither first not second anonymous object so we can not use filtering
// RenderedMessage : Hello World

Case 2:
Log.Debug("Hello World {@AnonymousParam1}", new { Param1 = "Test1" }, new { Param2 = "Test2" })
// Logs both params as expected
// RenderedMessage : Hello World { Param1: "Test1" }
// AnonymousParam1.Param1 : "Test1"
// __1.Param2 : "Test2"

**Please describe the expected behavior?**

Log.Debug("Hello World", new { Param1 = "Test1" }, new { Param2 = "Test2" })
// RenderedMessage : Hello World
// __1.Param1 : "Test1"
// __2.Param2: "Test2"

As a workaraund we could always add some stub parameter like "{stub}" to the message template, but I hope that a better solution exists.

To summarize, our goal is to collect information from parameters which aren't presented in the message template when the template doesn't contain any { } braces in it.

Thank you in advance.

**Comment by tsimbalar** [comment]
Hi,

thanks for filing an issue. The behavior you observe is *by design* and is the expected behavior.

If you want to attach extra properties that are not part of the *message template*, you should use one of the `.ForContext(...)` overloads (see https://github.com/serilog/serilog/wiki/Writing-Log-Events#correlation ). 

In your case, that would mean : 
```csharp
Log.ForContext("First", new { Param1 = "Test1" })
   .ForContext("Second", new { Param2 = "Test2" })
   .Debug("Hello World");
```

It turns out I asked a similar question on Stack Overflow back in the days :-D https://stackoverflow.com/questions/30637416/how-can-i-add-a-named-property-that-is-not-part-of-the-message-template 

There is also a [great series of articles](https://nblumhardt.com/2016/06/structured-logging-concepts-in-net-series-1/) by @nblumhardt that I would recommend, and in particular, this post covers *context enrichment* : https://nblumhardt.com/2016/08/context-and-correlation-structured-logging-concepts-in-net-5/ 

I hope this helps !

**Comment by Alfredovec** [comment]
Thank you!
